src/Data/Replicator/Linear.hs

```diff
@@ -0,0 +1,15 @@
+{-# LANGUAGE MagicHash #-}
+{-# OPTIONS_GHC -Wno-dodgy-exports #-}
+
+module Data.Replicator.Linear
+  ( Replicator,
+    next,
+    next#,
+    extract,
+    elim,
+    module Data.Replicator.Linear.Internal.Instances,
```

Comment:
Also export the other function `map`, `pure`, etc… they don't hurt. And that lets the programmer call them explicitly rather than through the

Reply:
I don't. Maybe it has non-instances in it, despite the name?

Reply:
Nice catch. Indeed, there are non-instance things exported. Maybe we should write an explicit export list for this module?

Reply:
As you feel like. Probably not in this PR, though.

You may have to write explicit export list for any non-internal module, anyway, just so that you can add Haddock sections in them.

Reply:
> Also export the other function map, pure, etc… they don't hurt. And that lets the programmer call them explicitly rather than through the

I don't really see the point. That could create conflicts if the programmer import the module unqualified, couldn't it?
I'm going to do it anyway (we could always revert the change). Should we also export `consume` (I don't think so, but again, I don't have enough hindsight)?